src/main/java/io/vertx/core/parsetools/impl/RecordParserImpl.java

```diff
@@ -42,6 +42,8 @@
   private Handler<Buffer> eventHandler;
   private Handler<Void> endHandler;
   private Handler<Throwable> exceptionHandler;
+  private boolean parsing;
```

Comment:
this new flag that seems to be there to avoid reentrancy is actually not tested, you need to add a test for it

Reply:
Done